### PR TITLE
Changing Name of file and adding doctests in file.

### PR DESCRIPTION
**Pull request by AasheeshLikePanner**
### Describe your change:
Change the File and Added doctests in maths/binary_exponentiation_2.py
Fixes Issue https://github.com/TheAlgorithms/Python/issues/9478

* [ ] Add an algorithm?
* [x] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms include at least one URL that points to Wikipedia or another similar explanation.
* [x] If this pull request resolves one or more open issues then the description above includes the issue number(s) with a [closing keyword](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue): "Fixes #ISSUE-NUMBER".
#9478

### CODE REVIEW

File: DIRECTORY.md [hunk 1]
```diff
@@ -548,7 +548,7 @@
   * [Basic Maths](maths/basic_maths.py)
   * [Binary Exp Mod](maths/binary_exp_mod.py)
   * [Binary Exponentiation](maths/binary_exponentiation.py)
-  * [Binary Exponentiation 2](maths/binary_exponentiation_2.py)
+  * [Binary Exponentiation 2](maths/binary_multiplication.py)
```
Comment (tianyizheng02) [inline DIRECTORY.md:551]: Please don't modify DIRECTORY.md. This file is auto-generated with a script.

File: maths/binary_multiplication.py [hunk 2]
```diff
@@ -23,6 +46,12 @@ def b_expo(a: int, b: int) -> int:
     return res
 
 
+if __name__ == "__main__":
+    import doctest
+
+    doctest.testmod()
```
Comment (tianyizheng02) [inline maths/binary_multiplication.py:52]: The main block should be placed at the bottom of the file.

File: maths/binary_multiplication.py [hunk 3]
```diff
@@ -23,6 +46,12 @@ def b_expo(a: int, b: int) -> int:
...
+
+
 def b_expo_mod(a: int, b: int, c: int) -> int:
```
Comment (tianyizheng02) [inline maths/binary_multiplication.py:55]: This function also needs doctests

**Review by tianyizheng02 — CHANGES_REQUESTED** [review]

**Comment by tianyizheng02** [comment]
??? Ok then

**Comment by AasheeshLikePanner** [comment]
Can you please check this one[Pr](https://github.com/TheAlgorithms/Python/pull/9513)

**Pull request closed without merging**